core/config.py

```diff
@@ -207,6 +211,26 @@ def database_url(cls):
         logging.info("Connecting to database: %s" % url_obj.__to_string__())
         return url
 
+    # TODO: We can remove this method once basic token authentication is fully deployed.
+    @classmethod
+    def basic_token_auth_is_enabled(cls) -> bool:
+        """Is basic token authentication enabled?
+
+        Return False, if the variable is unset or is an empty string.
+        Raises CannotLoadConfiguration, if the setting is invalid.
+        :raise CannotLoadConfiguration: If the setting contains an unsupported value.
+        """
+        try:
+            return ansible_boolean(
```

Comment:
Instead of defining a function to do this, I wonder if we would be better off leaning on Pydantic to do the parsing. I think we could accomplish the same thing with: 

```python
    @classmethod
    def basic_token_auth_is_enabled(cls) -> bool:
        """Is basic token authentication enabled?
        Return False, if the variable is unset or is an empty string.
        Raises CannotLoadConfiguration, if the setting is invalid.
        :raise CannotLoadConfiguration: If the setting contains an unsupported value.
        """
        try:
            return parse_obj_as(bool, os.environ.get(cls.BASIC_TOKEN_AUTH_ENABLED_ENVVAR))
        except (ValidationError) as e:
            raise CannotLoadConfiguration(
                f"Invalid value for {cls.BASIC_TOKEN_AUTH_ENABLED_ENVVAR} environment variable."
            ) from e
```

Which would parse the env var as a pydantic boolean https://docs.pydantic.dev/latest/usage/types/booleans/
> a str which when converted to lower case is one of '0', 'off', 'f', 'false', 'n', 'no', '1', 'on', 't', 'true', 'y', 'yes'

Reply:
Made a slight edit there to the code in my comment because I realized I wasn't handling the unset case in my first reply.

Reply:
@jonathangreen That's a clever and interesting idea. It works right now because `False` is the default, but it might cause some confusion if we were to decide at some point switch to a default of True. In that case, making a naive change the default and subsequently setting the environment variable to a falsy value would result in a value of True. It might lead to some programming errors down the line.

I also thought it would be helpful to have a function feature flags going forward, since I don't think this is the last time we'll see the need. And it has tests, so it's pretty easy to reason about how it will behave.

Another option would be to replace part of the logic in the current `ansible_boolean` with the `pydantic.parse_obj_as` approach above.

What do you think?

Reply:
I'm okay with whatever you choose, including just leaving it as is.

Reply:
Okay. Thanks!